das/dasRpcClient.go

```diff
@@ -61,24 +68,38 @@ func NewDASRPCClient(target string, signer signature.DataSignerFunc, maxStoreChu
 		return nil, fmt.Errorf("max-store-chunk-body-size %d doesn't leave enough room for chunk payload", maxStoreChunkBodySize)
 	}
 
+	url, err := url.Parse(target)
+	if err != nil {
+		return nil, err
+	}
+
 	return &DASRPCClient{
-		clnt:      clnt,
-		url:       target,
-		signer:    signer,
-		chunkSize: uint64(chunkSize),
+		clnt:       clnt,
+		url:        target,
+		signer:     signer,
+		chunkSize:  uint64(chunkSize),
+		metricName: metricsutil.CanonicalizeMetricName(url.Hostname()),
```

Comment:
It looks like CanonicalizeMetricName doesn't filter out periods, which don't seem to be an allowed Prometheus metric name but are of course common in URLs.

Reply:
I think it does replace them. The regex is picking characters that don't match the range (which doesn't include `.`) and replacing them with `_`. 

```
// Prometheus metric names must contain only chars [a-zA-Z0-9:_]
func CanonicalizeMetricName(metric string) string {
	invalidPromCharRegex := regexp.MustCompile(`[^a-zA-Z0-9:_]+`)
	return invalidPromCharRegex.ReplaceAllString(metric, "_")

}
```

If you look at the Nova dashboard, first panel, you can see the dots in domains have been replaced with `_`.